build.ts

```diff
@@ -84,43 +92,233 @@ async function makeThemeObject(
         tab_background_text: palette.overlay1.hex,
         tab_background_separator: palette.surface0.hex,
         bookmark_text: palette.text.hex,
         toolbar: palette.base.hex,
         toolbar_field: palette.surface0.hex,
         toolbar_field_text: palette.text.hex,
-        toolbar_field_highlight: palette[accent].hex,
+        toolbar_field_highlight: themeAccent,
         toolbar_field_highlight_text: palette.mantle.hex,
         toolbar_field_border: palette.mantle.hex,
         toolbar_field_focus: palette.surface0.hex,
         toolbar_field_text_focus: palette.text.hex,
-        toolbar_field_border_focus: palette[accent].hex,
+        toolbar_field_border_focus: themeAccent,
         toolbar_top_separator: palette.surface0.hex,
         toolbar_bottom_separator: palette.surface0.hex,
         toolbar_vertical_separator: palette.surface0.hex,
         sidebar: palette.base.hex,
         sidebar_text: palette.text.hex,
-        sidebar_highlight: palette[accent].hex,
+        sidebar_highlight: themeAccent,
         sidebar_highlight_text: palette.mantle.hex,
         sidebar_border: palette.surface0.hex,
         popup: palette.surface0.hex,
         popup_text: palette.text.hex,
         popup_border: palette.base.hex,
-        popup_highlight: palette[accent].hex,
+        popup_highlight: themeAccent,
         popup_highlight_text: palette.mantle.hex,
         spaces_bg: palette.mantle.hex,
         tree_view_bg: palette.crust.hex,
         bg_color: palette.base.hex,
-        spaces_bg_active: palette[accent].hex,
-        button_primary_bg: palette[accent].hex,
+        spaces_bg_active: themeAccent,
+        button_primary_bg: themeAccent,
+        button_text: palette.crust.hex,
+        spaces_button: palette.crust.hex,
+        tree_pane_bg: palette.crust.hex,
+        tree_card_bg: palette.mantle.hex,
+        layout_bg_0: palette.base.hex,
+        layout_bg_1: palette.base.hex,
+        button_bg: palette.mantle.hex,
+        lwt_accent_color: palette.mantle.hex,
+        list_container_background_selected_current: palette.mantle.hex,
+        ab_cards_list_bg: palette.mantle.hex,
+        in_content_box_info_background: palette.mantle.hex,
+        calendar_view_toggle_bg: palette.mantle.hex,
+        calendar_view_toggle_hover_bg: palette.surface0.hex,
+        tabs_toolbar_bg: palette.mantle.hex,
+        color_gray_70: palette.mantle.hex,
+        color_gray_50: palette.surface1.hex,
+      },
+    },
+  };
+}
+
+async function makeDarkLightThemeObject(
+  name: string,
+  accent: string,
+  flavor: CatppuccinFlavor,
+) {
+  let lightAccent: string = "";
+  let darkAccent: string = "";
+
+  if (flavor.dark) {
+    flavor.colorEntries.map(([colorName, { hex }]) => {
+      if (colorName == accent) {
+        darkAccent = hex;
+      }
+    });
+    flavors.latte.colorEntries.map(([colorName, { hex }]) => {
+      if (colorName == accent) {
+        lightAccent = hex;
+      }
+    });
+  } else return NaN;
```

Comment:
Since this function isn't meant to operate on the latte flavor, we could add a clause to skip calling it for latte in the main for loop, and use `Omit` or something to make that clear here.
```suggestion
  let lightAccent: flavor.colors[accent].hex;
  let darkAccent: flavors.latte.colors[accent].hex;
```

Reply:
I did not put too much thought into this and worked with what `flavor` had to offer like the `dark` boolean but this could be done better without the maps agreed.